.github/workflows/workflow.yml

```diff
@@ -457,22 +531,19 @@ jobs:
         version="$($HOME/.linkerd version --client --short | tr -cd '[:alnum:]-')"
         bin/test-run $HOME/.linkerd linkerd-$version
     - name: CNI tests
-      if: github.ref == 'refs/heads/master' || startsWith(github.ref, 'refs/tags')
       run: |
         export TAG="$($HOME/.linkerd version --client --short)"
         go test -cover -race -v -mod=readonly ./cni-plugin/test -integration-tests
 
   cloud_cleanup:
-    if: always()
     name: Cloud cleanup
+    if: github.ref == 'refs/heads/master' || startsWith(github.ref, 'refs/tags')
```

Comment:
similar to the `kind_cleanup` job, do we want to keep the `always()`, to ensure `cloud_cleanup` is called regardless of what happens in `cloud_integration`?

Reply:
Yep I'll add it. #4001 will remove this job soon as a side note.